### PR TITLE
Use `git describe` based build artifact versioning

### PR DESCRIPTION
**Pull request by dsharp-pivotal**

### CODE REVIEW

**Comment by dsharp-pivotal** [comment]
We have tested the changes on the master pipeline and are not going to test the PR pipeline because it would break any pending builds until this gets merged. Since the changes are comparable to those on the master pipeline, we're fairly confident in them.

File: concourse/scripts/gpdb_src_tinc_packaging.bash [hunk 1]
```diff
@@ -1,12 +1,20 @@
-#!/bin/bash -l
+#!/bin/bash
 
 set -euxo pipefail
```
Comment (cjcjameson) [inline concourse/scripts/gpdb_src_tinc_packaging.bash]: I believe you have modified the comprehensive set of packaging bash scripts, spot on.

**Comment by dsharp-pivotal** [comment]
Re: S3 bucket versioning: Yes, the bucket stays versioned, and this works fine. The "version-in-filename" artifacts simply will have only one S3 version (ideally--in in rebuild they may get more, and the resource will get the latest version).

**Comment by cjcjameson** [comment]
+1

**Pull request merged**